Microsoft.HealthVault.Fhir/Transformers/ThingBaseToFhir.cs

```diff
@@ -22,88 +23,65 @@ namespace Microsoft.HealthVault.Fhir.Transformers
-        /// <returns>A FHIR observation based on the HealthVault thing</returns>
-        public static Observation ToFhir(this ThingBase thing)
+        /// <returns>A FHIR resource based on the HealthVault thing</returns>
+        public static Resource ToFhir(this ThingBase thing)
         {
-            // Find registered specific ToFhir method
             var transformerType = typeof(ThingBaseToFhir);
             var method = transformerType.GetRuntimeMethod("ToFhir", new Type[] { thing.GetType() });
 
             if (method != null && method.GetParameters()[0].ParameterType != typeof(ThingBase))
             {
-                return (Observation)method.Invoke(null, new object[] { thing });
+                return (Resource)method.Invoke(null, new object[] { thing });
             }
             else
             {
-                return thing.ToFhirInternal();
+                //TODO Should Observation be the default?
+                return thing.ToFhirInternal<Observation>();
             }
         }
 
-        internal static Observation ToFhirInternal(this ThingBase thing)
+        internal static T ToFhirInternal<T>(this ThingBase thing) where T : Resource, IExtendable, new()
         {
-            var observation = new Observation();
-            observation.Meta = new Meta();
+            var resource = new T();
 
-            observation.AddExtension(HealthVaultVocabularies.FlagsFhirExtensionName, new FhirString(thing.Flags.ToString()));
-            observation.AddExtension(HealthVaultVocabularies.StateFhirExtensionName, new FhirString(thing.State.ToString()));
+            resource.SetIdentity(thing.Key);
+            resource.CopyLastUpdatedFromAudit(thing.LastUpdated);
 
-            observation.Status = ObservationStatus.Final;
+            //Extensions
+            resource.AddFlagsAsExtension(thing.Flags);
+            resource.AddStateAsExtension(thing.State);
 
-            if (thing.Key != null)
-            {
-                if (thing.Key.Id != null)
-                {
-                    observation.Id = thing.Key.Id.ToString();
-                }
+            resource.DoTransforms(thing);
 
-                if (thing.Key.VersionStamp != null)
-                {
-                    observation.Meta.VersionId = thing.Key.VersionStamp.ToString();
-                }
-            }
+            return resource;
+        }
 
-            if (thing.Created != null)
-            {
-                if (thing.Created.Timestamp != null)
-                {
-                    observation.Issued = thing.Created.Timestamp;
-                }
-            }
+        /// <summary>
+        /// Do transforms from thing to existing resource
+        /// </summary>
+        /// <param name="resource">A Fhir resource to add transforms to</param>
+        /// <param name="thing">The healthvault thing to get tranforms from</param>
+        public static void DoTransforms(this Resource resource, ThingBase thing)
```

Comment:
>DoTransforms [](start = 27, length = 12)

Why are we adding these DoTransforms? This feels like we are repeating the pattern that we already have with ToFhir().   To address the issue with Resource we should probably use Generics in all ToFhir methods so we can let the caller specify what kind of FHIR resource we want to use to transform.